.design.md

```diff
@@ -0,0 +1,19 @@
+This is a Python-based service following the Triple Hexagonal Architecture pattern.
+It uses protocol/provider pairs and dependency injection mechanisms provided by the
+[hexkit](https://github.com/ghga-de/hexkit) library.
+
+### Typical operation
+
+This service doesn't have a REST API or use a database for anything.
+It's a straightforward service running a Kafka consumer that listens for one kind of event.
+Notification events are picked up by the consumer, validated against the Notification event schema, and sent to the Notifier module.
+The Notifier looks at the notification event details and determines what to do with it.
+Right now, this always means sending an email.
+The information is sent to the SMTP client, where a secure connection is established and the email is dispatched.
```

Comment:
Nice but it is more about giving an overview on the functionality and API of the service. Thus this might better go into the description?